### PR TITLE
Bresenham Line Drawing Algorithm

### LINKED ISSUE #762: Bresenham Line Drawing algorithm
this generates a line

### PR DESCRIPTION
**Pull request by robotjellyzone**
Fixes #762

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [N] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Addition Of Bresenham Line Drawing Algorithm

#### Languages Used:
- C
- Java
- Python

#### Files Added:
- Bresenham Line Algorithm.c
-  Line drawing algo.java
-  Bresenham Line Algo.py

### CODE REVIEW

**Comment by jainaman224** [comment]
I can't see any changes.

**Comment by robotjellyzone** [comment]
@jainaman224 can you please check it again :smile

**Comment by jainaman224** [comment]
It is necessary to add sample input output. Add a link to image for line as there is a use of graphics library.

File: Bresenham_Algo_Programs/Line drawing algo.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    
+    }
+
+ // draw a line from point xa,ya into xb,yb
+ public void bresenham_line(int xa, int ya, int xb, int yb) {
+
+  // if point xa, ya is on the right side of point xb, yb, change them
+  if ((xa - xb) > 0) { bresenham_line(xb, yb, xa, ya); return; }
+
+  // test inclination of line
```
Comment (tsah20) [inline Bresenham_Algo_Programs/Line drawing algo.java]: If you are adding comments then please try following the standard commenting practice. 
For methods mention the parameter, return type in  a multiline comment

File: Bresenham_Algo_Programs/Bresenham Line Algo.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+from graphics import *
+import time
```
Comment (tsah20) [inline Bresenham_Algo_Programs/Bresenham Line Algo.py]: add newline between the package declaration and the import statements

File: Bresenham_Algo_Programs/Line drawing algo.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package applet;
```
Comment (tsah20) [inline Bresenham_Algo_Programs/Line drawing algo.java]: Please add a newline between package and imports

File: Bresenham_Algo_Programs/Line drawing algo.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  // draw line
+  for (int i = 0; i <= xb -xa; i++) {
+   setpix(x, y);
+   x++;
+   sum -= Dy;
+   if (sum < 0) {y = y + prirastokDy; sum += Dx;}
+  }
+ }
+
+ public void bresteepline(int xc, int yc, int xd, int yd) {
```
Comment (tsah20) [inline Bresenham_Algo_Programs/Line drawing algo.java]: Please declare method names in camel casing

File: Bresenham_Algo_Programs/Bresenham Line Algorithm.c [hunk 5]
```diff
@@ -0,0 +1,53 @@
+#include<stdio.h>
+#include<conio.h>
+#include<graphics.h>
+#include<math.h>
+
+// Driver Function 
+int main()
+{
+	/* Variables Defined -> for coordinates xa, ya, xb, yb of line */
```
Comment (amaaniqbal) [inline Bresenham_Algo_Programs/Bresenham Line Algorithm.c]: 4 space indentation is not followed in these lines. Also, I see this in many other places. Please try updating them. :)

**Comment by jainaman224** [comment]
@robotjellyzone Are you still working on it?

**Comment by robotjellyzone** [comment]
Yes, will complete it soon

**Comment by somya-kapoor** [comment]
@robotjellyzone Please change the file name. (Bresenham_Algo_Programs/Bresenham Line Algo.py to Bresenham_Algo_Programs/Bresenham_Line_Algo.py )

**Comment by MastersAbh** [comment]
@robotjellyzone Still working on it ?

**Comment by robotjellyzone** [comment]
Yes

**Comment by MastersAbh** [comment]
Sure. Please reach out if you are facing any issue.

**Pull request closed without merging**